yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerCloud.java

```diff
@@ -221,12 +221,13 @@ private DockerSlave provisionWithWait(DockerSlaveTemplate template, Provisioning
         final DockerContainerLifecycle dockerContainerLifecycle = template.getDockerContainerLifecycle();
         final String imageId = dockerContainerLifecycle.getImage();
         final DockerComputerLauncher computerLauncher = template.getLauncher();
+        final String osType = this.getClient().infoCmd().exec().getOsType();
```

Comment:
I have not checked Swarm.   Basically, all I did was port the [existing pull request for docker-plugin](https://github.com/jenkinsci/docker-plugin/pull/465) into this codebase and test the result in my infrastructure.  I agree, it would be better to get the OS type from the image.  That way, it won't matter where the image is running.